compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -667,6 +673,18 @@ private void checkErrors(TypeDefinition unresolvedType, BLangType currentTypeNod
         }
     }
 
+    private void addFunctionType(BLangSimpleVariable varNode, SymbolEnv env) {
```

Comment:
Can't we re-use an existing method? This doesn't seem to handle rest parameters.